lua/neogit/buffers/commit_editor/init.lua

```diff
@@ -40,25 +39,21 @@ function M:open()
     modifiable = true,
     readonly = false,
     autocmds = {
-      ["BufWritePre"] = function()
-        written = true
-      end,
       ["BufUnload"] = function(o)
-        if written then
-          if
-            not config.values.disable_commit_confirmation
-            and not input.get_confirmation("Are you sure you want to commit?")
-          then
-            -- Clear the buffer, without filling the register
-            vim.api.nvim_buf_set_lines(o.buf, 0, -1, false, {})
-            vim.api.nvim_buf_call(o.buf, function()
-              vim.cmd("silent w!")
-            end)
-          end
+        if
+          not vim.bo.mod
+          and not config.values.disable_commit_confirmation
+          and not input.get_confirmation("Are you sure you want to commit?")
+        then
+          -- Clear the buffer, without filling the register
+          vim.api.nvim_buf_set_lines(o.buf, 0, -1, false, {})
+          vim.api.nvim_buf_call(o.buf, function()
+            vim.cmd("silent w!")
+          end)
         end
 
         if self.on_unload then
-          self.on_unload(written)
+          self.on_unload(vim.bo.mod)
```

Comment:
should be 
```lua
self.on_unload(not vim.bo.mod)
```

even though this should get the right buffer. If you want to narrow it down to no other possibility than the unloaded buffer, we could add `vim.bo[o.buf].mod`